lib/awful/tooltip.lua

```diff
@@ -405,6 +412,21 @@ function tooltip:set_preferred_positions(value)
     set_geometry(self)
 end
 
+--- Set a widget to display.
+-- @tparam widget widget The widget to display
+-- @tparam integer width The width of the tooltip
+-- @tparam integer height The height of the tooltip
+function tooltip:set_widget(widget, width, height)
```

Comment:
The size of the screen is wrong. The correct value would be the size of the screen minus the placement of the tooltip, i.e. only the "really remaining space". But then we have a dependency loop between placement and size.

Also, the following widgets/layouts cannot be used in this scenario since they just take all available space:
- checkbox
- imagebox (mostly (allow_resize))
- piechart
- progressbar
- slider
- flex
- systray (mostly (base size needs to be unset), but still: ewh, no!)

These would work (sometimes, if they don't contain something from the above list):
- textbox
- textclock
- align
- constraint
- fixed
- grid

So... basically only a textbox works in the scenario you describe. And even then it is easy to get a really long tooltip where multiple lines would be better.

(I'm ignoring containers since they mostly pass on the fit result of some other widget)